superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/EchartsTimeseries.tsx

```diff
@@ -272,15 +235,14 @@ export default function EchartsTimeseries({
         // @ts-ignore
         const globalModel = echartInstance.getModel();
         const model = getModelInfo(params.target, globalModel);
-        const seriesCount = globalModel.getSeriesCount();
-        const currentSeriesIndices = globalModel.getCurrentSeriesIndices();
         if (model) {
           const { name } = model;
-          if (seriesCount !== currentSeriesIndices.length) {
-            handleDoubleClickChange();
-          } else {
-            handleDoubleClickChange(name);
-          }
+          const legendState: LegendState = legendData
+            .map(datum => ({
+              [datum]: datum === name,
+            }))
+            .reduce((previous, current) => ({ ...previous, ...current }));
```

Comment:
If I understand correctly, you don't need to have additional `.map` iteration. just `.reduce` can cover this

```suggestion
            .reduce((previous, datum) => ({
               ...previous,
               [datum]: datum == name,
            }));
```